### PR TITLE
feat(android): add "google-services.json" support

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27745

**Test:**
1. Download the [kitchensink-v2](https://github.com/appcelerator/kitchensink-v2) app.
2. Build and run on Android.
3. Verify that the app builds and runs without issue.
4. Download [google-services.json](https://axwaysoftware-my.sharepoint.com/:u:/g/personal/jquick_axway_com/EU-gKkYY4wRFpVN4mrOcr3UBI8Ci9-trkF6V8ZdzmAfiVA?e=gMH5Ut). _(Only Axway employees can access this.)_
5. Copy the `google-services.json` file to: `./app/platform/android`
6. Build and run on Android.
7. In the log, verify that you see the following message.
```
[INFO]  [GRADLE] Parsing json file: /Users/<YourFolder>/kitchensink-v2/build/android/app/google-services.json
```

### CODE REVIEW

File: android/cli/commands/_build.js [hunk 1]
```diff
@@ -2244,6 +2244,13 @@ AndroidBuilder.prototype.generateAppProject = async function generateAppProject(
 		}
 	}
 
+	const googleServicesFile = path.join(this.projectDir, 'platform', 'android', 'google-services.json');
```
Comment (cb1kenobi) [inline android/cli/commands/_build.js]: The previous for-loop copies all files in the `platform/android` directory into the `build/android/app/src/main`. So after this new code, you would have a `google-services.json` file in both `build/android/app/src/main` and `build/android/app`. Do you need the file in both places? Maybe not a bad idea to update the `afs.copyDirSyncRecursive()` call above to pass in `ignoreFiles: [ 'google-services.json' ]`.

Comment (hansemannn) [reply android/cli/commands/_build.js]: I talked with @jquick-axway about it and he mentioned I'd be okay to keep it for a later usage. But it's also fine to ignore it if it's really not used otherwhere.

Comment (cb1kenobi) [reply android/cli/commands/_build.js]: I'm cool either way. Thanks and hope you're doing well! Cheers!

Comment (hansemannn) [reply android/cli/commands/_build.js]: Cool! Same to you man, missing the unofficial architecture hangouts 😄

**Comment by jquick-axway** [comment]
@hansemannn, can you add the below code to the top of our [app.build.gradle](https://github.com/appcelerator/titanium_mobile/blob/master/android/templates/build/app.build.gradle) template please? The idea is we want to automatically use the plugin if the `google-services.json` file exists since the plugin's main purpose is to read the file and turn it into "res" info. The below code should go just below our other `apply plugin: '...'` statement. Thanks!
```groovy
if (file("${projectDir}/google-services.json").exists()) {
	apply plugin: 'com.google.gms.google-services'
}
```

**Comment by hansemannn** [comment]
@jquick-axway Added it!

**Review by jquick-axway — APPROVED** [review]
CR: Pass

**Comment by ssekhri** [comment]
FR Passed
Mac OS: 10.15.4
SDK: 9.1.0.v20200413130540
Appc CLI: 8.0.0
JDK: 11.0.4
Node: 10.17.0
Studio: 6.0.0.202003181504
Device: Pixel3(v10.0) emulator

**Pull request merged**